presentation-api/controlling-ua/startNewPresentation_success-manual.html

```diff
@@ -39,11 +72,11 @@
 
                         // Check the instance of the connection
                         assert_true(connection instanceof PresentationConnection, 'The connection is an instance of PresentationConnection.');
-                    });
+                    }), function(e) { reject(e); });
+            });
```

Comment:
As often with these tests that mix promises and events, I'm struggling to find the right "recipe" for the test to be clear and simple. Having a `catch` function that calls `reject` works but does not "look nice" for instance.

In this case, I wonder if we could not benefit from the `EventWatcher` class that the test harness exposes and which allows to handle events through promises. Applied to this test, this could give:

```js
        promise_test(function (t) {
            var request = new PresentationRequest(presentationUrls);
            return request.start()
                .then(function (c) {
                    // Phase #1: Promise is resolved
                    connection = c;

                    // No more user input needed, re-enable timeout
                    t.step_timeout(function() {
                        t.force_timeout();
                        t.done();
                    }, 5000);

                    // Check the initial state of the presentation connection
                    assert_equals(connection.state, 'connecting', 'Check the initial state of the presentation connection.');
                    assert_true(!!connection.id, 'The connection ID is set.');
                    assert_true(typeof connection.id === 'string', 'The connection ID is a string.');
                    assert_true(connection instanceof PresentationConnection, 'The connection is an instance of PresentationConnection.');

                    var eventWatcher = new EventWatcher(t, request, 'connectionavailable');
                    return eventWatcher.wait_for('connectionavailable');
                })
                .then(function (evt) {
                    // Phase #2: "connectionavailable" event fired
                    assert_equals(connection, evt.connection, 'Both Promise from PresentationRequest() and a "connectionavailable" event handler receive the same presentation connection.');
                    
                    var eventWatcher = new EventWatcher(t, connection, 'connect');
                    return eventWatcher.wait_for('connect');
                })
                .then(function () {
                    // Phase #3: "connect" event fired
                    assert_equals(connection.state, 'connected', 'The state of the presentation connection is "connected" when a "connect" event fires.');
                });
        });
```

This code would be slightly less complete than yours: it ensures things happen in the right order, but e.g. will not fail if a browser fires a `connectionavailable` event, then resolves the start promise and then fires _another_ `connectionavailable` event for instance. That said, one advantage is that the different phases appear more clearly and in logical order in the code. Also, `EventWatcher` will make the test fail if other events get fired at the request or at the connection object.

What do you think?